### PR TITLE
replace if_let_chain with if_chain

### LINKED ISSUE #2150: Remove `if_let_chain` and use `if_chain` crate
I think we should not roll our own macro, but instead use the crate of the week: https://crates.io/crates/if_chain

### PR DESCRIPTION
**Pull request by durka**
Fixes #2150. Done mostly mechanically with a hacked-up version of @japaric's untry.

### CODE REVIEW

**Comment by durka** [comment]
The tests don't get very far on my machine due to some procedural macro BS, so I figured travis could have a crack at it.

**Comment by durka** [comment]
Rebased on #2153 temporarily.

**Comment by oli-obk** [comment]
I merged #2153, can you rebase over master?

File: clippy_lints/src/collapsible_if.rs [hunk 1]
```diff
@@ -100,43 +100,45 @@ fn check_if(cx: &EarlyContext, expr: &ast::Expr) {
...
-    if_let_chain! {[
-        let Some(inner) = expr_block(then),
-        let ast::ExprKind::If(ref check_inner, ref content, None) = inner.node,
-    ], {
-        if expr.span.ctxt() != inner.span.ctxt() {
-            return;
+    if_chain! {
+        if let Some(inner) = expr_block(then);
+        if let ast::ExprKind::If(ref check_inner, ref content, None) = inner.node;
+        then {
```
Comment (oli-obk) [inline clippy_lints/src/collapsible_if.rs]: I think I'd prefer it if we didn't have to use an additional indentation level. Maybe use something like

```rust
if_chain! {
   ...
then {
   ...
}}
```

Opinions?

Comment (durka) [reply clippy_lints/src/collapsible_if.rs]: I went with the indentation used in the crate's examples: https://docs.rs/if_chain/0.1.2/if_chain/

**Comment by durka** [comment]
Rebased.

File: clippy_lints/src/collapsible_if.rs (hunk 1 above)
Comment (oli-obk) [reply clippy_lints/src/collapsible_if.rs]: Ok, I can get behind staying with the official style

**Comment by durka** [comment]
I think I can fix that. At least for line comments it should be easy...

**Comment by durka** [comment]
I recovered the comments.

File: clippy_lints/src/new_without_default.rs [hunk 2]
```diff
@@ -117,40 +117,41 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for NewWithoutDefault {
             if decl.inputs.is_empty() && name == "new" && cx.access_levels.is_reachable(id) {
                 let self_ty = cx.tcx
                     .type_of(cx.tcx.hir.local_def_id(cx.tcx.hir.get_parent(id)));
-                if_let_chain!{[
-                                    same_tys(cx, self_ty, return_ty(cx, id)),
```
Comment (durka) [inline clippy_lints/src/new_without_default.rs]: Indentation is wrong here, but I didn't cause that...

**Comment by oli-obk** [comment]
Awesome! Another code duplication bites the dust

**Pull request merged**